openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -1507,7 +1507,96 @@ It is up to the Credential Issuer whether to update both the signature and the c
 
 # Privacy Considerations
 
-TBD
+When [@!RFC9396] is used, the Privacy Considerations of that specification also apply.
+
+The privacy principles of [@ISO.29100] should be adhered to.
+
+## User Consent
+
+The Credential Issuer should obtain the End-User's consent before issuing Credential(s)
+to the Wallet. It should be made clear to the End-User what information is being included in the
+Credential(s) and for what purpose.
+
+## Minimum Disclosure
+
+To ensure minimum disclosure and prevent Verifiers from obtaining claims unnecessary for the
+transaction at hand, when issuing Credentials that are intended to be created once and then used a
+number of times by the End-User, the Credential Issuers and the Wallets SHOULD implement
+Credential formats that support selective disclosure, or consider issuing a separate Credential
+for each user claim.
+
+## Storage of the Credentials
+
+To prevent a leak of End-User data, especially when it is signed, which risks revealing private data of End-Users to
+third parties, systems implementing this specification SHOULD be designed to minimize the amount
+of End-User data that is stored. All involved parties SHOULD store Verifiable Credentials
+containing privacy-sensitive data only for as long as needed, including in log files. Any logging of
+End-User data should be carefully considered as to whether it is necessary at all. The time logs are retained
+for should be minimized.
+
+After Issuance, Credential Issuers SHOULD NOT store the Issuer-signed Credentials if they
+contain privacy-sensitive data. Wallets SHOULD store Credentials only in encrypted form, and,
+wherever possible, use hardware-backed encryption. Credentials that are no longer valid to present should be deleted as soon as
+possible.
+
+## Correlation 
+
+### Unique Values Encoded in the Credential
+
+Issuance/presentation or two presentation sessions by the same End-User can be linked on the basis of
+unique values encoded in the Credential (End-User claims, identifiers, Issuer signature, etc.) either by colluding Issuer/Verifier or Verifier/Verifier pairs, or by the same Verifier.
+
+To prevent these types of correlation, Credential Issuers and Wallets SHOULD use
+methods, including but not limited to the following ones:
+
+* Issue a batch of Credentials to enable the usage a unique Credential per presentation or per Verifier using Batch Credential Endpoint defined in (#batch-credential-endpoint). This only helps with Verifier/Verifier unlinkability.
+* Use cryptographic schemes that can provide non-correlation.
```

Comment:
```suggestion
* Use cryptographic schemes that can provide non-correlation.
* Issuers should discard credential identifiers, such as the signature or subject key, after issuance to prevent Issuer/Verifier collusion.
```

Reply:
what's the definition of "credential identifiers"

this is needed for a better understanding of your proposed change

GDPR and national regulation prevents this, since the signed documents must be retained for years since are citizen's personal data handled by the issuers (for legacy infrastructurs id token are retained from 2 to 5 years). There might be other requirements about national regulation where the storage of a signed artifact is needed to resolve any disputes in the future

Reply:
I think this is what Jacob meant..?
```suggestion
* Use cryptographic schemes that can provide non-correlation.
* Discard values that can be used in Issuer/Verifier collusion, such as Issuer's signature or cryptographic key material to which an issued credential was bound to.
```

Reply:
Yep exactly

Reply:
> GDPR and national regulation prevents this, since the signed documents must be retained for years since are citizen's personal data handled by the issuers (for legacy infrastructurs id token are retained from 2 to 5 years). There might be other requirements about national regulation where the storage of a signed artifact is needed to resolve any disputes in the future

@peppelinux is this true for all kinds of credentials or only "identity" credentials? If this is only for identity credentials then I think my point is still valid.

Besides, if we believe a mechanism is good for privacy then I think it is worthy of being in a privacy considerations section, even if it is ruled out by regulations. Regulations change over time, so this consideration could become more relevant.

Reply:
> Yes, except it only applies to issuers. I don't think we want the wallet discarding those values :)

I added `(only applies to Issuers)`.

Reply:
```suggestion
* Use cryptographic schemes that can provide non-correlation.
* Credential Issuers specifically SHOULD discard values that can be used in collusion with a Verifier to track a user, such as the Issuer's signature or cryptographic key material to which an issued credential was bound to.
```

Alternative suggestion to the language above, has the same affect IMO.